nvdaHelper/localWin10/oneCoreSpeech.cpp

```diff
@@ -126,85 +419,111 @@ wstring OcSpeech::getVoices() {
 			voices += L"|";
 		}
 	}
-	return voices;
+	return SysAllocString(voices.c_str());
 }
 
-// We use BSTR because we need the string to stay around until the caller is done with it
-// but the caller then needs to free it.
-// We can't just use malloc because the caller might be using a different CRT
-// and calling malloc and free from different CRTs isn't safe.
-BSTR __stdcall ocSpeech_getVoices(OcSpeech* instance) {
-	return SysAllocString(instance->getVoices().c_str());
-}
-
-hstring OcSpeech::getCurrentVoiceId() {
-	return synth.Voice().Id();
-}
-
-const wchar_t* __stdcall ocSpeech_getCurrentVoiceId(OcSpeech* instance) {
-	return instance->getCurrentVoiceId().c_str();
-}
-
-void OcSpeech::setVoice(int index) {
-	synth.Voice(synth.AllVoices().GetAt(index));
-}
-
-void __stdcall ocSpeech_setVoice(OcSpeech* instance, int index) {
-	instance->setVoice(index);
-}
-
-hstring OcSpeech::getCurrentVoiceLanguage() {
-	return synth.Voice().Language();
-}
 
-const wchar_t* __stdcall ocSpeech_getCurrentVoiceLanguage(OcSpeech* instance) {
-	return instance->getCurrentVoiceLanguage().c_str();
+bool isUniversalApiContractVersion_(const int major, const int minor) {
+	constexpr auto contract_name{ L"Windows.Foundation.UniversalApiContract" };
+	return ApiInformation::IsApiContractPresent(
+		winrt::hstring{ contract_name },
+		major,
+		minor
+	);
 }
 
-double OcSpeech::getPitch() {
-	return synth.Options().AudioPitch();
-}
 
-double __stdcall ocSpeech_getPitch(OcSpeech* instance) {
-	return instance->getPitch();
-}
-
-void OcSpeech::setPitch(double pitch) {
-	synth.Options().AudioPitch(pitch);
+void preventEndUtteranceSilence_(std::shared_ptr<winrtSynth> synth) {
+	// By default, OneCore speech appends a  large annoying chunk of silence at the end of every utterance.

```

Comment:
Fix additional space
```suggestion
	// By default, OneCore speech appends a large annoying chunk of silence at the end of every utterance.
```